lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -627,12 +629,14 @@ def __init__(self, model_name_or_path, device=None):
         self.device = device
         self._model_name_or_path = model_name_or_path
 
-        from transformers import AutoImageProcessor, DeformableDetrForObjectDetection
+        lock = fasteners.InterProcessLock(_DETR_LOCK_FILE)
+        with lock:
+            from transformers import AutoImageProcessor, DeformableDetrForObjectDetection
 
-        LogTime("loading_model", point=True)
-        with LogTime("load_model", log_start=True):
-            self.processor = AutoImageProcessor.from_pretrained(model_name_or_path)
-            self.model = DeformableDetrForObjectDetection.from_pretrained(model_name_or_path).to(self._get_device())
+            LogTime("loading_model", point=True)
+            with LogTime("load_model", log_start=True):
+                self.processor = AutoImageProcessor.from_pretrained(model_name_or_path)
```

Comment:
I'm not sure where, but you should have code that does something like os.walk(f"{os.environ['HOME']}/.cache/torch_extensions") and deletes files called 'lock'

Reply:
Done.